deployment/docker/get_filter_messages/filter_msg_retriever.py

```diff
@@ -0,0 +1,91 @@
+# Python Imports
+import argparse
+import logging
+import urllib
+import requests
+import socket
+import time
+from concurrent.futures import ProcessPoolExecutor, as_completed
+from typing import Dict, List, Optional
+
+logging.basicConfig(level=logging.INFO)
+
+
+def resolve_dns(address: str) -> str:
+    start_time = time.time()
+    name, port = address.split(":")
+    ip_address = socket.gethostbyname(name)
+    elapsed = (time.time() - start_time) * 1000
+    logging.info(f"{address} DNS Response took {elapsed} ms")
+    logging.info(f"Talking with {address}, ip address: {ip_address}")
+
+    return f"{ip_address}:{port}"
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(description="Waku filter message retriever")
+    parser.add_argument('-c', '--contentTopic', type=str, help='Content topic', default="/my-app/1/dst/proto")
+    parser.add_argument('-n', '--numNodes', type=int, help='Number of filter nodes to get messages from', default=1)
+
+    return parser.parse_args()
+
+
+def fetch_all_messages(base_url: str, headers: Dict, address: str) -> Optional[List[str]]:
+    response = requests.get(base_url, headers=headers)
+    if response.status_code != 200:
+        logging.error(f"Error fetching data: {response.status_code}")
+        logging.error(response.text)
+        return None
+
+    data = response.json()
+    messages = [message_data['payload'] for message_data in data]
```

Comment:
Microoptimization, in case you wanna go for it (and fits, ofc). Generally it's better to rely on generators unless you need to read the list multiple times.

Reply:
First thing I want from it is the `len`, so I don't think it would fit in this case.